core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1220,9 +1221,8 @@ public InsertTableHandle beginRefreshMaterializedView(Session session, TableHand
         List<ConnectorTableHandle> sourceConnectorHandles = sourceTableHandles.stream()
                 .map(TableHandle::connectorHandle)
                 .collect(Collectors.toList());
-        sourceConnectorHandles.add(tableHandle.connectorHandle());
```

Comment:
nit: this could be separate commit, but let's leave it

Reply:
Happy to separate it out if you think it's important. Otherwise, please let me know if there's anything I can do before merging this. Thanks!